bin/citrea/tests/bitcoin_e2e/tests/sequencer_commitments.rs

```diff
@@ -0,0 +1,306 @@
+        for _ in 0..MIN_SOFT_CONF_PER_COMMITMENT - 1 {
+            sequencer.client.send_publish_batch_request().await;
+        }
+        sequencer
+            .wait_for_l2_height(MIN_SOFT_CONF_PER_COMMITMENT - 1, None)
+            .await;
+
+        da.generate(5, None).await?;
+
+        let finalized_height = da.get_finalized_height().await?;
+
+        for height in initial_height..finalized_height {
+            let mut blobs = da.get_relevant_blobs_from_block(height).await?;
+
+            for mut blob in blobs.drain(0..) {
+                let data = blob.full_data();
+
+                assert_eq!(data, &[] as &[u8]);
+            }
+        }
+
+        // Publish one more L2 block and send commitment
+        sequencer.client.send_publish_batch_request().await;
+        sequencer
+            .wait_for_l2_height(MIN_SOFT_CONF_PER_COMMITMENT, None)
+            .await;
+
+        // Wait for blob tx to hit the mempool
+        da.wait_mempool_len(1, None).await?;
+
+        // Include commitment in block and finalize it
+        da.generate(5, None).await?;
+
+        let start_l2_block = 1;
+        let end_l2_block = 4;
+
+        self.check_sequencer_commitment(sequencer, da, start_l2_block, end_l2_block)
+            .await?;
+
+        for _ in 0..MIN_SOFT_CONF_PER_COMMITMENT {
+            sequencer.client.send_publish_batch_request().await;
+        }
+        sequencer
+            .wait_for_l2_height(MIN_SOFT_CONF_PER_COMMITMENT * 2, None)
+            .await;
+
+        // Wait for blob tx to hit the mempool
+        da.wait_mempool_len(1, None).await?;
+        // Include commitment in block and finalize it
+        da.generate(5, None).await?;
+
+        let start_l2_block = end_l2_block + 1;
+        let end_l2_block = start_l2_block + 4;
+
+        self.check_sequencer_commitment(sequencer, da, start_l2_block, end_l2_block)
+            .await?;
+
+        Ok(())
+    }
+}
+
+impl SequencerSendCommitmentsToDaTest {
+    async fn check_sequencer_commitment(
+        &self,
+        sequencer: &Sequencer,
+        da: &BitcoinNode,
+        start_l2_block: u64,
+        end_l2_block: u64,
+    ) -> Result<()> {
+        let finalized_height = da.get_finalized_height().await?;
+
+        // Extract and verify the commitment from the block
+        let mut blobs = da.get_relevant_blobs_from_block(finalized_height).await?;
+
+        assert_eq!(blobs.len(), 1);
+
+        let mut blob = blobs.pop().unwrap();
+
+        let data = blob.full_data();
+
+        let commitment = DaData::try_from_slice(data).unwrap();
+
+        matches!(commitment, DaData::SequencerCommitment(_));
+
+        let DaData::SequencerCommitment(commitment) = commitment else {
+            panic!("Expected SequencerCommitment, got {:?}", commitment);
+        };
+
+        let mut batch_receipts = Vec::new();
+
+        for i in start_l2_block..=end_l2_block {
+            batch_receipts.push(
```

Comment:
name this vec soft_confirmations

we use this name now, batch receipts was a name unsuitable for our rollup actually, we misused it